code/mob/living/silicon/ai.dm

```diff
@@ -255,6 +254,22 @@ or don't if it uses a custom topopen overlay
 	..()
 	src.turn_it_back_on()
 
+/mob/living/silicon/ai/emag_act(mob/user, obj/item/card/emag/E)
+	if (src.dismantle_stage >= 2)
```

Comment:
should respect `src.emaggable` var like `robot` since that var is on the parent `silicon`